### DISCUSSION TITLE
H96 Max X4 (S905X4 100MB) install issue

### DISCUSSION THREAD

**Discussion by gitisgreat2023** [devmfc/debian-on-amlogic#35] [Q&A]
H96 Max 8K Ultra HD X4 with S905X4 and 100MB ethernet doesn't boot.
I tried x96x4, x96x4_gigabit, s905x4_generic and s905x4_generic_gigabit
Both with a SD card and an USB stick.

Currently CoreELEC is installed, which runs perfectly. However, I prefer a normal linux distro so this project is perfect! I would love to be able to use it.

It either shows a frozen H96Max screen logo or it goes to the Android Recovery menu or it shows a black screen. From my experience with other tv boxes this happens when the wrong image is used.
Because of the black screen (HDMI might not be working) I put an ethernet cable in and checked for new devices on the network: no, weren't.

I'm out of ideas currently. Has anyone successfully booted the H96 Max S905X4 with this distro?

A second question: how do I get rid of the Android (boot loader) which is still on the EMMC drive? Ideally only the linux is on the disk, without remainings of Android and CoreELEC.

**Comment by devmfc** [comment]
I would expect s905x4_generic to just work. 
You can try two things:
- restore stock android firmware and try again (bootloader has changed for some reason?)
- Connect a serial cable to the board to see what is going on.

**Reply by gitisgreat2023** [reply]
"Connect a serial cable to the board to see what is going on." What do I have to do there? There's no serial port on the box...
Restore stock? How do I do that?

I'm completely stuck right now, any insights would be greatly appreciated.

**Reply by gitisgreat2023** [reply]
Okay, I managed to flash again Android on it, using the Amlogic USB burning tool.
I just ordered a H96 Max X3... that should work. The difference is only the GPU, but I will run it headless anyway.

 @devmfc what's the way to have a linux only install on the internal flash driev, on the tv box I ordered? I don't want Android on it, just only Linux. I saw in an another issue that first multiboot is needed, before the install to emmc is possible?

**Reply by devmfc** [reply]
Does your S905X4 box boot after restoring stock firmware?

Yes, first `./aml-multiboot-setup.sh`, reboot to see if it works (optional). After that install to emmc with `./aml-install-to-emmc.sh
`.

**Comment by gitisgreat2023** [comment]
No, I tried a lot of thinks, also checking with LAN whether it might just have a HDMI issue.
It's a H96Max 8k Ultra HD X4 with 4GB/64GB. I flashed 905x4_l002_ZX_100M_4g-20230810.2140.img which works like a charm.

I'm looking forward to my X3, hope the installation runs smooth. I also have an Armbian box. The main difference is just that your distro is a standard Linux with the right kernel, vs armbian a more different flavour, right?
Performance wise there's no big difference, is there?

**Reply by devmfc** [reply]
Armbian is also Debian/Ubuntu based. Biggest difference is, I think, that my images only use Debian/Ubuntu apt sources and have less modifications. And my images have specific DTB's for some tvboxes (so more on board hardware will work like vfd display, wifi, BT). I don't expect there is any performance difference between Armbian and my images. 

Have you opened your X4? I'm beginning to get really curious why it doesn't work... If so, can you post a picture of boh sides of the board?

**Reply by gitisgreat2023** [reply]
No, I didn't open it and actually not want so... 
I actually saved the dmesg of CoreELEC before I flashed Android on it. Would that help you instead?

**Comment by gitisgreat2023** [comment]
@devmfc the s905x boots successfully, however only detects 3.5 of the 4GB of RAM (box=h906maxx3). What should I do to fix this?
This surprises me, I haven't seen this issue for years. Another Armbian Box has no issues with 4GB.
Is that some issue that other devices reserve memory?

**Reply by devmfc** [reply]
AFAIK, you can"t. It is just reserved memory. But if you'll find an image that has more available memory, please let me know, 

I would love to see the DTB to see where the difference is.

**Reply by jojkol** [reply]
I have the same box as his, I burned the image into the sd card, but after I logged into debian and plugged the keyboard into the usb port of the box, none of the apt commands worked, and I couldn't install any software

**Reply by devmfc** [reply]
Run _apt-get update_ first?

**Reply by jojkol** [reply]
> 先运行_apt-get update_？

I followed your instructions and ran the apt-get update command. It prompted that the repository address was inrelease temporaru fallure resolving. I got errors when installing various software, such as Reading package lists... Done
Building dependency tree... Done
Reading state information... Done, etc. I suspect it's related to the apt source. I tried changing the apt source, but it always failed. I really can't think of a solution. How can I change the apt source?

**Reply by jojkol** [reply]
> 先运行_apt-get update_？

I wrote the image to the SD card, restarted the box and entered the Ubuntu or Debian system through the AV port. After entering this image system, I found that the network port of the box could not be used, but when the box was an Android system, the network was normal.
